hw11/mailroom.py

```diff
@@ -0,0 +1,96 @@
+
+
+def mailroom(userinput='thank you'):
+    fulldonorlist = [["Sydney Decoto", "Big Bird", "John Doe", "Jane Doe",
+                      "Cher"], [[20], [300, 400], [1000], [600, 50],
+                    [20, 30, 40]], []]
+
+    while userinput != 'quit':
+        userinput = input("""\nWelcome to Mailroom Madness\n
+Choose from the following:\n\nT - Send a (T)hank You\n
+R - Create a (R)eport\n\nquit - Quit the program\n\n""")
+
+        if userinput == 'T':
+            fulldonorlist = thankyou(userinput, fulldonorlist)
+        elif userinput == 'R':
+            createreport(fulldonorlist)
+        elif userinput != 'quit':
+            print('invalid entry')
+
+
+# Prepare the thank you email
+def thankyou(userinput, fulldonorlist):
+    while userinput != 'quit':
+        userinput = input("""\nPlease enter a name, or choose from the following:\n
+list - Print a list of previous donors\n
+quit - Return to main menu\n\n""")
+        check = False
+        # Initialize the donor list
+        donorlist = fulldonorlist[0]
+        # If the user types list, print the names
+        if userinput == 'list':
+            print(donorlist)
+            userinput = input("Enter Donor Name: ")
+            check = userinput in fulldonorlist[0]
+            if check:
+                ind = fulldonorlist[0].index(userinput)
+            else:
+                fulldonorlist[0].append(userinput)
+        elif userinput != 'quit':
+            # append list of donors with new name
+            # donorlist.append(userinput)
+            check = userinput in fulldonorlist[0]
+            if check:
```

Comment:
This same code appears above in the userinput == 'list' block. Try not to repeat yourself whenever possible.